File: vsf/bundles/org.eclipse.vjet.vsf.vjo.kernel/src/org/eclipse/vjet/vsf/NvBinding.java
/*******************************************************************************
 * Copyright (c) 2012 eBay Inc. and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     eBay Inc. - initial API and implementation
 *******************************************************************************/
package org.eclipse.vjet.vsf;

import org.eclipse.vjet.dsf.javatojs.anno.AJsProxy;
import org.eclipse.vjet.dsf.dap.proxy.NativeJsProxy;
import org.mozilla.mod.javascript.Scriptable;
import org.eclipse.vjet.dsf.javatojs.anno.AExclude;
import org.eclipse.vjet.dsf.dap.proxy.INativeJsFuncProxy;
import org.eclipse.vjet.dsf.dap.proxy.NativeJsFuncProxy;
import org.eclipse.vjet.dsf.dap.proxy.NativeJsTypeRef;
import org.eclipse.vjet.dsf.javatojs.anno.AJavaOnly;
import org.eclipse.vjet.vsf.IBinding;
import org.eclipse.vjet.vsf.Message;

//NativeJsProxy for vjo.dsf.NvBinding.js
@org.eclipse.vjet.dsf.resource.utils.CodeGen("NativeJsProxyGenerator")
@AJsProxy
public class NvBinding extends NativeJsProxy implements IBinding {

    /** for framework use only */
    @AExclude
    public NvBinding(Scriptable nativeObj){
        super(nativeObj);
    }

    /** internal use only */
    protected NvBinding(Object ...args){
        super(args);
    }

    public NvBinding() {
        super();
    }

    public String serialize(Message message) {
        return callWithName("serialize", String.class, message);
    }

    public void deserialize(Message message) {
        callWithName("deserialize", message);
    }

    @AJavaOnly
    public static final NativeJsTypeRef<NvBinding> prototype = NativeJsTypeRef.get(NvBinding.class);

    public final INativeJsFuncProxy<NvBinding> serialize = NativeJsFuncProxy.create(this, "serialize");

    public final INativeJsFuncProxy<NvBinding> deserialize = NativeJsFuncProxy.create(this, "deserialize");
}

File: vsf/bundles/org.eclipse.vjet.vsf.vjo.kernel/src/org/eclipse/vjet/vsf/NvBinding.js
/*******************************************************************************
 * Copyright (c) 2012 eBay Inc. and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     eBay Inc. - initial API and implementation
 *******************************************************************************/
/**
 * Name value pair binding
 * serialize implemented only
 */
vjo.ctype('org.eclipse.vjet.vsf.NvBinding') //< public
.satisfies('org.eclipse.vjet.vsf.IBinding')
.needs('org.eclipse.vjet.vsf.Service')
.needs('org.eclipse.vjet.vsf.Enc')
//> needs org.eclipse.vjet.vsf.Message
.protos({
 	//>public String serialize(Message message) 
 	serialize : function(message){
 		message.rawRequest = org.eclipse.vjet.vsf.Service.generateReqParams(message) + org.eclipse.vjet.vsf.Enc.encodeURIComponent(message.request);
		return message.rawRequest;
	},
 	//>public void deserialize(Message message) 
 	deserialize : function( message){
 	}
})
.endType();